src/lib/ObjectAssertion.ts

```diff
@@ -3,7 +3,7 @@ import { isDeepStrictEqual } from "util";
 
 import { Assertion } from "./Assertion";
 
-export type JSObject = Record<keyof any, unknown>;
+export type JSObject = Record<keyof any, any>;
```

Comment:
This change fixes issues with type inference in generic constraints. The type `unknown` is too specific to be part of a constraint